pkg/fftm/transaction_events_handler.go

```diff
@@ -58,12 +58,12 @@ func (eh *ManagedTransactionEventHandler) HandleEvent(_ context.Context, e apity
 			Transaction: &confirmations.TransactionInfo{
 				TransactionHash: e.Tx.TransactionHash,
 				Receipt: func(ctx context.Context, receipt *ffcapi.TransactionReceiptResponse) {
-					if err := eh.TxHandler.HandleTransactionReceiptReceived(ctx, e.Tx.ID, receipt); err != nil {
+					if err := eh.TxHandler.HandleTransactionReceiptReceived(ctx, e.Tx, receipt); err != nil {
 						log.L(ctx).Errorf("Receipt for transaction %s at nonce %s / %d - hash: %s was not handled due to %s", e.Tx.ID, e.Tx.TransactionHeaders.From, e.Tx.Nonce.Int64(), e.Tx.TransactionHash, err.Error())
 					}
 				},
 				Confirmations: func(ctx context.Context, notification *apitypes.ConfirmationsNotification) {
-					if err := eh.TxHandler.HandleTransactionConfirmations(ctx, e.Tx.ID, notification); err != nil {
+					if err := eh.TxHandler.HandleTransactionConfirmations(ctx, e.Tx, notification); err != nil {
```

Comment:
I'm worried about this change.
Passing a pointer to a long term background callback implies an invalid assumption about the use of this callback.

While there might be policy engines for whom this is a safe thing to do, we cannot rely on that in the generic interface.

Reply:
I reverted this change in my review so this can merge with the important change of providing access to the filters.